samples/kctf-conf/base/Makefile

```diff
@@ -149,14 +150,54 @@ clean:
 
 .deploy: .deployment .cluster-config .network
 
-.network: .gen/k8s | .cluster-config
+define CERTIFICATE_TEMPLATE
+apiVersion: "networking.gke.io/v1beta1"
+kind: "ManagedCertificate"
+metadata:
+  name: "kctf-certificate"
+spec:
+  domains:
+endef
+
+define INGRESS_TEMPLATE
+apiVersion: "networking.k8s.io/v1beta1"
+kind: "Ingress"
+metadata:
+  name: "https"
+  annotations:
+    networking.gke.io/managed-certificates: "kctf-certificate"
+spec:
+  backend:
+    serviceName: "chal"
+    servicePort: 1
+endef
+
+export CERTIFICATE_TEMPLATE
+export INGRESS_TEMPLATE
+
+.https: .cluster-config
+ifeq ($(HTTPS), true)
+ifneq ($(DNS), true)
+	$(error 'HTTPS=true but DNS=false. Need DNS to get certificate.')
+endif
+ifeq ($(DOMAIN_NAME),)
+        $(error 'HTTPS=true but DOMAIN_NAME is not set.')
+endif
+	(echo "$${CERTIFICATE_TEMPLATE}"; echo '    -' "$(CHALLENGE_NAME).$(DOMAIN_NAME)") | kubectl apply -f -
+	echo "$${INGRESS_TEMPLATE}" | kubectl apply -f -
+	kubectl annotate "ingress/https" "external-dns.alpha.kubernetes.io/hostname=${CHALLENGE_NAME}-http.${DOMAIN_NAME}" --overwrite
```

Comment:
btw, you don't need to annotate Ingresses: DNS entry is updated for them automatically using host form spec

Reply:
hmm, but where does it get the host name from?

Reply:
From `spec.rules.host` within Ingress. If it's unspecified like now in Makefile it means that it matches any `*` hostname.

For dns-controller to be able to see domain we need to do:

```yaml
spec:
  rules:
  - host: ${CHALLENGE_NAME}-http.${DOMAIN_NAME}"
    http:
      paths:
      - backend:
          serviceName: "chal"
          servicePort: 1
```

Reply:
hmm, does this mean we maybe would only need a single Ingress, and the route stuff based on the hostname to different ports?

say if your challenge has 3 containers, or 3 ports, we could expose 8080 on `${CHALLENGE_NAME}-8080.${DOMAIN_NAME}` and 8081 on `${CHALLENGE_NAME}-8080.${DOMAIN_NAME}` and so forth. Always using port 80/443, but mapping to different internal services.

This would probably save money, because public IPs and LoadBalancers are expensive, so having a single one per challenge is better (it would actually be better if we could have a single TCP load balancer for the whole CTF and just different port numbers, and then a single HTTPS load balancer and different hostnames).

Reply:
hmm seems that this isn't possible to do cross-namespace ingress https://github.com/kubernetes/kubernetes/issues/17088

I guess we can still do this for multiple HTTP ports on a single task, though.

Reply:
Yes, for using it for multiple ports is a good idea.

Reply:
ok, since the current `chal.conf` model (with booleans) doesn't know of the multiple ports, I think we should move this idea to CRD. I'll file a bug for this.

Reply:
https://github.com/google/kctf/issues/112